### ISSUE TITLE
Promise dataSource gets stringified and used as URL

### ISSUE THREAD

**Issue by JohanX** [bencripps/react-redux-grid#85]
When using the grid with a promise dataSource as specified in the [docs](https://github.com/bencripps/react-redux-grid/blob/master/docs/PROVIDING_DATA.md#if-data-is-provided-as-a-function-it-must-return-a-promise) the default pager doesn't work.
In [src/actions/plugins/pager/PagerActions.js:112](https://github.com/bencripps/react-redux-grid/blob/master/src/actions/plugins/pager/PagerActions.js#L112) the dataSource is being passed as URL to the Api method which in turn results in a stringified version of whatever code was provided as dataSource.

**Comment by bencripps** [comment]
Could you provide the props you're providing to grid please?

**Comment by bencripps** [comment]
I was able to repro -- working on a fix now.

**Comment by JohanX** [comment]
The dataSource looks like this:
``` js       
const dataSource = () => {
            return new P((success) => promiseThatRetrievesDataFromServer().then((response) => {
                return success(response.payload);
            }));
        };
```
And the pager in the plugins object looks like this:
```js
            PAGER: {
                enabled: true,
                pagingType: 'remote',
                pagerComponent: false
            },
```
When clicking the Next button on the pager (which is enabled only when total number of rows is greater than the returned number in a single call), the Api method requests the string version of the dataSource function mentioned above. This string is generated here `src/util/api.js:67`

**Comment by bencripps** [comment]
I'm going to add this function to the docs, so that it will be easier to understand how to construct a promise that returns the expected shaped object. However, there was a bug in identifying which type of `dataSource` was provided. Please get the lastest version of master, and see if this solves your problem. If it does -- I will make a new release.

Thanks for posting the issue!

```js

export const examplePromiseDataSource = function getData({
    pageIndex, pageSize
}) {
    return new Promise((resolve) => {
        const request = new XMLHttpRequest();

        const config = {
            method: 'GET',
            route: 'http://react-redux-grid.herokuapp.com/getfakeData'
        };

        if (pageIndex) {
            config.route = `${config.route}?pageIndex=${pageIndex}&pageSize=${pageSize}`; // eslint-disable-line max-len
        }

        request.open(config.method, config.route, true);

        request.addEventListener(
            'load', (res) => {
                const response = JSON.parse(res.target.responseText);

                // if you have more data than is being shown
                // ensure you return a total, so the pager knows
                // what paging actions are possible

                resolve({
                    data: response.data,
                    total: response.total
                });
            }
        );

        request.send(config.data || null);
    });
};
```

**Comment by JohanX** [comment]
I can confirm that it is now working as described in the latest doc change, including the passing of pageSize and pageIndex parameters.

**Comment by bencripps** [comment]
Will be released as 5.1.6
